src/wallet/walletdb.h

```diff
@@ -59,6 +63,11 @@ class CHDChain
         READWRITE(this->nVersion);
         READWRITE(nExternalChainCounter);
         READWRITE(masterKeyID);
+        if (this->nVersion >= SUPPORT_WATCHONLY_VERSION) {
+            READWRITE(isWatchOnly);
+            if(isWatchOnly)
+                READWRITE(masterPubKey);
```

Comment:
Wait... I was using the extended private key instead. Though, I guess it should not crash in this case.